engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java

```diff
@@ -78,49 +99,73 @@ public abstract class BaseTable extends LivenessArtifact
      */
     protected final String description;
 
-    // Attribute support
-    protected final ConcurrentHashMap<String, Object> attributes = new ConcurrentHashMap<>();
+    // Attribute support, set via ensureField with ATTRIBUTES_UPDATER if not initialized
+    volatile Map<String, Object> attributes;
 
     // Fields for DynamicNode implementation and update propagation support
-    private transient boolean refreshing;
-    private transient Condition updateGraphProcessorCondition;
-    private transient Collection<Object> parents;
-    private transient SimpleReferenceManager<ShiftObliviousListener, WeakSimpleReference<ShiftObliviousListener>> childListenerReferences;
-    private transient SimpleReferenceManager<ShiftObliviousListener, WeakSimpleReference<ShiftObliviousListener>> directChildListenerReferences;
-    private transient SimpleReferenceManager<TableUpdateListener, WeakSimpleReference<TableUpdateListener>> childShiftAwareListenerReferences;
-    private transient volatile long lastNotificationStep;
-    private transient volatile long lastSatisfiedStep;
-    private transient boolean isFailed;
-
-    public BaseTable(@NotNull final TableDefinition definition, @NotNull final String description) {
+    private volatile boolean refreshing;
```

Comment:
I'm not sure why that's a relevant consideration. We want to ensure visibility to other threads, which may not other-wise have a happens-before relationship that ensures they read the result of the write to refreshing rather than the initial state. Since this would result in wrong behavior, it seems safer to make this more cautious.